osu.Game/Screens/Play/Player.cs

```diff
@@ -208,10 +208,12 @@ private void load(AudioManager audio, OsuConfigManager config, APIAccess api)
                 new HotkeyRetryOverlay
                 {
                     Action = () => {
-                        //we want to hide the hitrenderer immediately (looks better).
-                        //we may be able to remove this once the mouse cursor trail is improved.
-                        RulesetContainer?.Hide();
-                        Restart();
+                        if (IsCurrentScreen) {

```

Comment:
could you please invert both of these conditionals (ie. `if (!IsCurrentScreen) return;`)?